packages/nimble-components/docs/css-guidelines.md

```diff
@@ -126,6 +126,92 @@ For example:
 
 Useful reference: [When do the :hover, :focus, and :active pseudo-classes apply?](https://bitsofco.de/when-do-the-hover-focus-and-active-pseudo-classes-apply/)
 
+## Hierarchical Cascade Layers with @layer
+
+If you find yourself constantly overriding css classes when conditionally changing them based on the value of an html attribute, the [@layer css at-rule](https://developer.mozilla.org/en-US/docs/Web/CSS/@layer) should be used to declare a cascade layer and define the order of precedence / hierarchy in css classes. 
+
+An example of constant overriding:
+```css
+.content:hover {
+    border-color: ${borderHoverColor};
+}
+
+:host([some-attribute='some-value']) .content {
+    border-color: ${actionRgbPartialColor};
+}
+
+:host([some-attribute='some-value']) .content:hover {
+    border-color: ${borderHoverColor};
+}
+
+:host([some-attribute='some-other-value']) .content {
+    border-color: ${actionRgbPartialColor};
+}
+
+:host([some-attribute='some-other-value']) .content:hover {
+    border-color: ${borderHoverColor};
+}
+```
+In this example, the border-color class has to be constantly overridden back to `borderHoverColor` in `:host .content:hover` because `:host .content` changes the color to `actionRgbPartialColor` and overrides the `.content:hover` css class.
+
+This example instead uses hierarchical cascade layers with @layer:
+```css
+@layer base, hover, ...
+
+@layer base {
+    :host([some-attribute='some-value']) .content {
+        border-color: ${actionRgbPartialColor};
+    }
+
+    :host([some-attribute='some-other-value']) .content {
+        border-color: ${actionRgbPartialColor};
+    }
+}
+
+@layer hover {
+    .content:hover {
+        border-color: ${borderHoverColor};
+    }
+}
+```
+In this example, using @layer gives the `content:hover` non-host class precedence to apply to all `:host` classes in the `hover` state.
+
+States should flow from plain base -> hover -> focus -> active -> error -> disabled -> top (which overrides all others)
```

Comment:
I think it's worth mentioning (after this line?) that selectors that are not in a layer take precedence over all layers, so if you use layers, you should put all selectors in a layer.